document helm chart upgrades after k8s version upgrades (#3019)
wireapp/wire-server: docs/src/how-to/administrate/kubernetes/upgrade-cluster/index.md

## Kubeadm

Please refer to the *official documentation:* [Upgrading kubeadm clusters](https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade/)

# Troubleshooting problems arising after kubernetes cluster upgrades

## Helm and kubernetes API changes

If you upgrade to new versions of kubernetes while wire-server is deployed, you may find that after that version update, deploying a new version of wire-server or nginx-ingress-services (or another helm chart we provide) using `helm update` or `helmfile apply/sync` gives an error like this:

> Error: UPGRADE FAILED: current release manifest contains removed kubernetes api(s) for this kubernetes version and it is therefore unable to build the kubernetes objects for performing the diff. error from kubernetes: unable to recognize "": no matches for kind "Ingress" in version "extensions/v1beta1"

> What's happening here is that some kubernetes API versions [get deprecated](https://kubernetes.io/docs/reference/using-api/deprecation-guide/) and eventually removed. While we strive to keep maximum compatibility of kubernetes versions in our helm charts, that's not sufficient when doing k8s upgrades while wire-server helm charts are in use: you need to tell a helm release about the difference in API version.

Reviewer: ```suggestion
What's happening here is that some kubernetes API versions [get deprecated](https://kubernetes.io/docs/reference/using-api/deprecation-guide/) may potentially have been removed. While we strive to keep maximum compatibility of kubernetes versions in our helm charts, that's not sufficient when doing k8s upgrades while wire-server helm charts are in use: you need to tell a helm release about the difference in API version.
```

> What's happening here is that some kubernetes API versions [get deprecated](https://kubernetes.io/docs/reference/using-api/deprecation-guide/) and may potentially have been removed. While we strive to keep maximum compatibility of kubernetes versions in our helm charts, that's not sufficient when doing k8s upgrades while wire-server helm charts are in use: you need to tell a helm release about the difference in API version.

Reviewer: ```suggestion
What's happening here is that some kubernetes API versions [get deprecated](https://kubernetes.io/docs/reference/using-api/deprecation-guide/) may potentially have been removed. While we strive to keep maximum compatibility of kubernetes versions in our helm charts, that's not sufficient when doing k8s upgrades while wire-server helm charts are in use: you need to tell a helm release about the difference in API version.
```
